test/new_relic/gem_notifications_tests.rb

```diff
@@ -0,0 +1,73 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+
+require_relative '../test_helper'
+require_relative '../../.github/workflows/scripts/slack_gem_notifications/notifications_methods'
+
+class GemNotifications < Minitest::Test
+  def invalid_gem_response
+    response = MiniTest::Mock.new
+    response.expect :success?, false
+  end
+
+  def valid_gem_response
+    response = MiniTest::Mock.new
+    response.expect :success?, true
+  end
+
+  def http_get_response
+    [{"created_at": "2022-03-30T16:15:29.083Z", "platform": "ruby"},
+      {"created_at": "2021-03-30T16:15:29.083Z", "platform": "java"},
+      {"created_at": "2020-03-30T16:15:29.083Z", "platform": "ruby"}]
+  end
+
+  def test_check_for_updates_if_empty
+    response = MiniTest::Mock.new
+    response.expect :empty?, true
+    [].stub :empty?, response do
+      error = assert_raises(Exception) { check_for_updates([]) }
+      assert_equal("Nothing to see here! watched_gems cannot be empty", error.message)
+    end
+  end
+
+  def valid_gem_name
+    response = valid_gem_response()
+    HTTParty.stub :get, response do
+      assert_nil is_valid_gem?("puma!")
+    end
+  end
+
+  def test_invalid_gem_name
+    response = invalid_gem_response()
+    HTTParty.stub :get, response do
+      assert_nil is_valid_gem?("TrexRawr!")
+    end
+  end
+
+  def test_get_gem_info_returns_array
+    versions = gem_versions(http_get_response())
+    assert_instance_of Array, versions
+  end
+
+  def test_get_gem_info_max_size
+    versions = gem_versions(http_get_response())
+    assert_equal true, versions.size <= 2
+  end
+
+  def test_gem_updated
+    assert_equal true, gem_updated?([{"created_at" => "#{Time.now}"}])
+    assert_equal false, gem_updated?([{"created_at" => "1993-06-11T17:31:14.298Z"}])
+  end
+
+  def test_send_bot_input_size
+    assert_raises(ArgumentError) { send_bot() }
+    assert_raises(ArgumentError) { send_bot("tyrannosaurus") }
+    HTTParty.stub :post, nil do
+      assert_nil send_bot("tyrannosaurus", [{"number" => "1.2.3"}])
+    end
+    HTTParty.stub :post, nil do
+      assert_nil send_bot("tyrannosaurus", [{"number" => "1.2.3"}, {"number" => "4.5.6"}, {"number" => "7.8.9"}])
```

Comment:
This assertion is interesting because to me it indicates that the `send_bot()` method currently doesn't something I would think we don't want it to do in that it will happily accept a version array with anything other than exactly 2 entries. Is that desired?

If not, you can guard against any array size other than 2 with something like:

```ruby
def send_bot(gem_name, versions)
  raise 'Expected exactly 2 version numbers (newest and previous) in the 'versions' array' unless versions.size == 2
 
  ...
```

or even:

```ruby
def send_bot(gem_name, newest_version, previous_version)
  ...
```

and then add the appropriate assertions to check the functionality.

Reply:
Originally I imagined a world where a watched_gem was on V1. But I'm seeing how there would be no case where we'd send a bot without the previous version, unless we somehow knew about the gem before it was even released. 

I'll refactor this.